Scripts/Source/minai_Sex.psc

```diff
@@ -1194,3 +1033,278 @@ string Function GetFactionsForActor(actor akTarget)
 
   return ret
 EndFunction
+
+String Function JoinActorArray(Actor[] actorArray, string separator = ",")
+  String result = ""
+  int index = 0
+  int size = actorArray.Length
+
+  ; Loop through the array using a while loop
+  while (index < size)
+      Actor currentActor = actorArray[index]
+      if (currentActor)
+          ; Append the actor's name to the result string
+          result += Main.GetActorName(currentActor)
+
+          ; If it's not the last actor, append a comma
+          if (index < size - 1)
+              result += separator
+          endif
+      endif
+      
+      index += 1
+  endwhile
+  
+  return result
+EndFunction
+
+String Function JoinStringArray(string[] strArr, string separator = ",")
+  String result = ""
+  int index = 0
+  int size = strArr.Length
+
+  ; Loop through the array using a while loop
+  while (index < size)
+      string str = strArr[index]
+      if (str)
+          ; Append the actor's name to the result string
+          result += str
+
+          ; If it's not the last actor, append a comma
+          if (index < size - 1)
+              result += separator
+          endif
+      endif
+      
+      index += 1
+  endwhile
+  
+  return result
+EndFunction
+
+; type - is db operation: startthread | scenechange | speedchange | end | clean
+;  startthread - when new thread started
+;  scenechange - when scene changed
+;  speedchange - when speed changed
+;  end - on thread end
+;  clean - clean db of all threads recorded in table
+; framework - ostim or sexlab
+function UpdateThreadTable(string type, string framework = "ostim", int ThreadID = -1)
+  actor[] actors
+  string sceneId
+  
+  ; !!!Min to review sexlab implementation
+  if(framework == "sexlab")
+    sslThreadController controller = slf.GetController(ThreadID)
+  
+    if (controller.Stage==1) 
+      LoadSexlabDescriptions()
+    EndIf
+    
+    actors = slf.GetController(ThreadID).Positions
+    sceneId = controller.Animation.FetchStage(controller.Stage)[0]
+  else
+    actors = OThread.GetActors(ThreadID)
+    sceneId = OThread.GetScene(ThreadID)
+  endif
+
+  actor[] maleActors = PapyrusUtil.ActorArray(0)
+  actor[] femaleActors = PapyrusUtil.ActorArray(0)
+  int i = 0
+  int count = actors.Length
+  while i < count
+    actor currActor = actors[i]
+    int currSex = currActor.GetActorBase().GetSex()
+    
+    if(currSex == 0)
+      maleActors = PapyrusUtil.PushActor(maleActors, currActor)
+    elseif(currSex == 1)
```

Comment:
What about actors of type 2 (sex = other)? Let's just count them as female.

Reply:
is there other gender? :) I thought has only these options
```papyrus
; -1 - None
; 0 - Male
; 1 - Female
```

Reply:
It's safe to assume that any >0 should be treated as female

Reply:
wrote this function to handle more genders, will it work?

```papyrus
int function GetGender(actor targetActor)
  int sex = targetActor.GetActorBase().GetSex()
  
  ; sex 1+ is considered female
  if(sex >= 1)
    return 1
  endif

  return sex
endfunction
```

Reply:
Looks good